migrations/v4.js

```diff
@@ -0,0 +1,189 @@
+  mutateContent('Tutor - add globals if missing', async (content) => {
+    const course = content.find(({ _type }) => _type === 'course');
+    if (!_.has(course, '_globals._extensions._tutor')) _.set(course, '_globals._extensions._tutor', {});
+    courseTutorGlobals = course._globals._extensions._tutor;
+    return true;
+  });
+  mutateContent('Tutor - add globals hideFeedback', async (content) => {
+    courseTutorGlobals.hideFeedback = 'Hide feedback';
+    return true;
+  });
+  checkContent('Tutor - check globals _tutor attribute', async content => {
+    if (courseTutorGlobals === undefined) throw new Error('Tutor - globals _tutor invalid');
+    return true;
+  });
+  checkContent('Tutor - check globals hideFeedback attribute', async content => {
+    const isValid = courseTutorGlobals.hideFeedback === 'Hide feedback';
+    if (!isValid) throw new Error('Tutor - globals hideFeedback invalid');
+    return true;
+  });
+
+  // Update components
+  mutateContent('Tutor - add _tutor to components if missing', async (content) => {
+    components = content.filter(({ _component }) => _component);
+    components.forEach(component => {
+      if (!_.has(component, '_tutor')) _.set(component, '_tutor', {});
+    });
+    return true;
+  });
+  mutateContent('Tutor - add _tutor._type to components', async (content) => {
+    components.forEach(component => { _.set(component._tutor, '_type', 'notify'); });
+    return true;
+  });
+  mutateContent('Tutor - add _tutor._classes to components', async (content) => {
+    components.forEach(component => { _.set(component._tutor, '_classes', ''); });
+    return true;
+  });
+  mutateContent('Tutor - add _tutor._hasNotifyBottomButton to components', async (content) => {
+    components.forEach(component => { _.set(component._tutor, '_hasNotifyBottomButton', false); });
+    return true;
+  });
+  mutateContent('Tutor - add _tutor._button to components', async (content) => {
+    components.forEach(component => { _.set(component._tutor, '_button', {}); });
+    return true;
+  });
+  mutateContent('Tutor - add _tutor._button.text to components', async (content) => {
+    components.forEach(component => { _.set(component._tutor._button, 'text', '{{_globals._extensions._tutor.hideFeedback}}'); });
+    return true;
+  });
+  mutateContent('Tutor - add _tutor._button.ariaLabel to components', async (content) => {
+    components.forEach(component => { _.set(component._tutor._button, 'ariaLabel', '{{_globals._extensions._tutor.hideFeedback}}'); });
+    return true;
+  });
+  checkContent('Tutor - check for _tutor component attribute', async content => {
+    const isValid = components.every(({ _tutor }) => _tutor !== undefined);
+    if (!isValid) throw new Error('Tutor - components _tutor invalid');
+    return true;
+  });
+  checkContent('Tutor - check for _tutor._type on components', async content => {
+    const isValid = components.every(({ _tutor }) => _tutor?._type === 'notify');
```

Comment:
I think we can remove the optional chaining from these checks as you've already checked for tutor previously.